weave

```diff
@@ -847,7 +849,7 @@ case "$COMMAND" in
             -p $PORT:$CONTAINER_PORT/tcp -p $PORT:$CONTAINER_PORT/udp \
             -v /var/run/docker.sock:/var/run/docker.sock \
             -e WEAVE_PASSWORD \
-            $WEAVE_DOCKER_ARGS $IMAGE -iface $CONTAINER_IFNAME -port $CONTAINER_PORT -name "$PEERNAME" -nickname "$(hostname)" $IPRANGE "$@")
+            $WEAVE_DOCKER_ARGS $IMAGE -iface $CONTAINER_IFNAME -port $CONTAINER_PORT -name "$PEERNAME" -nickname "$(hostname)" $IPRANGE $ARGS)
```

Comment:
You can quote the arguments you want to keep (using a bit of sed to protect any quotes that occur), and then `eval set` to put that back into the positional parameters, like so:

```
# set with no args does something different, so we need to eliminate that case
if [ $# -gt 0 ] ; then
   ARGS=
   while [ $# -gt 0 ] ; do
       case "$1" in
           -password)
               shift
               ;;
           *)
               ARGS="$ARGS '$(echo "$1" | sed "s|'|'\"'\"'|g")'"
               shift
               ;;
       esac
   done

   eval "set -- $ARGS"
fi

echo "$@"
echo $#
```

Alternatively, I think it can be also done using computed variable names (e.g. save the argument values to $arg0, $arg1, $arg2 etc., then repeatedly do `set -- "$@" "${arg$n}"` to put them back into the positional parameters).